File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        freq = [[] for i in range(len(nums) + 1)]
        res = []

        for n in nums:
            count[n] = 1 + count.get(n, 0)
        for c in count:
            freq[count[c]].append(c)

        # print(count)
        # print(freq)
        for i in range(len(freq) - 1, -1, -1):
            for num in freq[i]:
                res.append(num)
                
            if len(res) == k:
                return res
        




        

File: Data Structures & Algorithms/permutation-string/submission-1.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        vector<int> target_count(26, 0);
        vector<int> count(26, 0);

        for (int i = 0; i < s1.size(); i++){
            target_count[(int)s1[i] - (int)'a'] += 1;
        }
        
        int l = 0;
        for (int r = 0; r < s2.size(); r++){
            count[(int)s2[r] - (int)'a'] += 1;
            if (r - l + 1 > s1.size()){
                count[(int)s2[l] - (int)'a'] -= 1;
                l++;
            }
            if (count == target_count) return true;
        }
        return false;
    }
};
